### PR TITLE
Updated the Az Databricks cli to 2024-05-01 api version.

### PR DESCRIPTION
**Pull request by v-vyedavelly**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az databricks workspace create/update: Add --access-connector to associate an Access Connector Resource with workspace.
az databricks workspace create/update: Add --default-storage-firewall to set default storage firewall configuration information on workspace.
az databricks workspace create/update: Add --enhanced-security-compliance to enable the Enhanced Security and Compliance on workspace.
az databricks workspace delete: Add --force-deletion to delete all data on Uc enabled workspace.

### General Guidelines

- [Y] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [Y] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [Y] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/databricks/setup.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 # TODO: Confirm this is the right version number you want and it matches your
 # HISTORY.rst entry.
-VERSION = '0.10.2'
+VERSION = '0.11.0'
```
Comment (AllyW) [inline src/databricks/setup.py:19]: please update version as cli extension versioning bot suggested.

for more info, please refer to this doc: https://github.com/Azure/azure-cli/blob/dev/doc/extensions/versioning_guidelines.md

**Comment by v-vyedavelly** [comment]
@yonzhan , All the PR checks were passed, the PR is ready to review and merge.

**Comment by kairu-ms** [comment]
Please fix the following breaking change. It seems miss an option name.

> ❌Azure CLI Extensions Breaking Change Test
> > ❌databricks
> > 
> > 
> > 
> > rule
> > cmd_name
> > rule_message
> > suggest_message
> > 
> > 
> > 
> > 
> > ❌ [1010 - ParaPropUpdate](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1010.md)
> > databricks access-connector update
> > cmd `databricks access-connector update` update parameter `user_assigned_identities`: updated property `options` from `['--identities', '--user-assigned-identities']` to `['--user-assigned-identities']`
> > please change property `options` from `['--user-assigned-identities']` to `['--identities', '--user-assigned-identities']` for parameter `user_assigned_identities` of cmd `databricks access-connector update`
> >

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**